.github/workflows/checks.yml

```diff
@@ -24,22 +27,31 @@ jobs:
         uses: jwlawson/actions-setup-cmake@4f73d30c2fc44a9466a3ed890fb8db7a7b554302
         with:
           cmake-version: '3.16.x'
-      - name: Set up Python 3.9.12
-        uses: actions/setup-python@v4
+      - name: Set up Python 3.9.12 and devnet
+        uses: actions/setup-python@v4.5.0
         with:
           python-version: 3.9.12
-
-      - name: Install devnet
-        run: |
+          cache: 'pip'
+      - run: |
           python -m pip install --upgrade pip
           pip install -r requirements.txt
 
       - name: Setup Gradle
         uses: gradle/gradle-build-action@v2
+        with: 
+          cache-read-only: false
 
       - name: Lint
         run: ./gradlew :lib:lintKotlin
 
+      - name: Cache cairo (rust) compiled deps
+        uses: actions/cache@v3
+        with:
+          path: cairo/target/debug
+          key: ${{ runner.os }}-cairo-target-debug-${{ hashFiles('cairo/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-cairo-target-debug-
```

Comment:
probably missing Cargo.lock hash, right?

Reply:
It's meant actually as a fallback to at least partially restore dependencies. I guess we could remove it entirely since we do not check whether the rust deps have changed, only that they are present in `compileContracts`.